runtime-modules/project-token/src/lib.rs

```diff
@@ -0,0 +1,611 @@
+        who: T::AccountId,
+        amount: T::Balance,
+    ) -> DispatchResult {
+        if amount.is_zero() {
+            return Ok(());
+        }
+
+        Self::ensure_token_exists(token_id).map(|_| ())?;
+
+        // == MUTATION SAFE ==
+
+        // increase token issuance
+        TokenInfoById::<T>::mutate(token_id, |token_data| {
+            token_data.current_total_issuance =
+                token_data.current_total_issuance.saturating_add(amount)
+        });
+
+        if AccountInfoByTokenAndAccount::<T>::contains_key(token_id, &who) {
+            AccountInfoByTokenAndAccount::<T>::mutate(token_id, &who, |account_data| {
+                account_data.free_balance = account_data.free_balance.saturating_add(amount)
+            });
+        } else {
+            AccountInfoByTokenAndAccount::<T>::insert(
+                token_id,
+                &who,
+                AccountDataOf::<T> {
+                    free_balance: amount,
+                    reserved_balance: T::Balance::zero(),
+                },
+            );
+        }
+
+        Self::deposit_event(RawEvent::TokenAmountDepositedInto(token_id, who, amount));
+        Ok(())
+    }
+
+    /// Mint `amount` into valid account `who`
+    /// for specified token `token_id`
+    ///
+    /// Preconditions:
+    /// - `token_id` must exists
+    /// - `who` must exists
+    ///
+    /// Postconditions:
+    /// - free balance of `who` is increased by `amount`
+    /// if `amount` is zero it is equivalent to a no-op
+    fn deposit_into_existing(
+        token_id: T::TokenId,
+        who: T::AccountId,
+        amount: T::Balance,
+    ) -> DispatchResult {
+        if amount.is_zero() {
+            return Ok(());
+        }
+
+        // ensure token validity
+        Self::ensure_token_exists(token_id).map(|_| ())?;
+
+        // ensure account id validity
+        Self::ensure_account_data_exists(token_id, &who).map(|_| ())?;
+
+        // == MUTATION SAFE ==
+
+        // increase token issuance
+        TokenInfoById::<T>::mutate(token_id, |token_data| {
+            token_data.current_total_issuance =
+                token_data.current_total_issuance.saturating_add(amount)
+        });
+
+        AccountInfoByTokenAndAccount::<T>::mutate(token_id, &who, |account_data| {
+            account_data.free_balance = account_data.free_balance.saturating_add(amount)
+        });
+
+        Self::deposit_event(RawEvent::TokenAmountDepositedInto(token_id, who, amount));
+
+        Ok(())
+    }
+
+    /// Burn `amount` of token `token_id` by slashing it from `who`
+    ///
+    /// Preconditions:
+    /// - `token_id` must exists
+    /// - `who` must exists
+    ///
+    /// Postconditions:
+    /// - free balance of `who` is decreased by `amount` or set to zero if below existential
+    ///   deposit
+    /// if `amount` is zero it is equivalent to a no-op
+    fn slash(token_id: T::TokenId, who: T::AccountId, amount: T::Balance) -> DispatchResult {
```

Comment:
This is best intended as a private method (and it will be used in the Bonding curve feature with high probability).
So I moved into the Module `impl` in b2726deded2bf71154e0e8e5e8ad01c4fbaaa257